torch/csrc/jit/passes/onnx/constant_fold.cpp

```diff
@@ -487,6 +487,14 @@ c10::optional<at::Tensor> runTorchBackendForOnnx(
     if (q > 1) {
       return c10::nullopt;
     }
+    // If the device of indices tensor is not the same with it of the input
+    // tensor, move it to the device of the input tensor
+    auto indices_val = node->inputs()[1];
+    if (indices_val->node()->kind() == onnx::Constant) {
```

Comment:
Is this check necessary?

Reply:
Hi @BowenBao thanks for the review! This check is needed as the issue is caused by module `constant`s being retrieved back as `cpu` tensors during constant folding. Therefore in this fix we only convert those constant tensor(s) to input device if 1. they are constant 2. they are on a different device than input tensors (most likely cpu vs cuda)

Reply:
Hmm for my own knowledge, what are cases when `indices_val` node kind is not onnx::Constant? I think at this point we are guaranteed `indices` a.k.a `inputTensorValues[1]` as tensor data is available to us. Is there other possibilities for `indices_val` nodekind?

And if the answer is yes, shouldn't we also cast if its device is not the same as `inputTensorValues[0]`'s?

Reply:
That makes sense. It is changed. Please take a look!